real_estate/models/real_estate_property.py

```diff
@@ -0,0 +1,119 @@
+    partner_id = fields.Many2one("res.partner", string = "Buyer", copy=False)
+    salesperson_id = fields.Many2one("res.users", string = "Salesman", default = lambda self: self.env.user)
+    tag_ids = fields.Many2many('real.estate.property.tag', string = "Tag")
+    offer_ids = fields.One2many('real.estate.property.offer', 'property_id', string = 'Offer', copy = False)
+    total_area = fields.Float(compute = "_compute_total_area", string = "Total Area (sqm)", readonly = True)
+    best_price = fields.Float(compute = "_compute_best_price", string = "Best Offer", readonly = True, default = 0, copy = False)
+    sold = fields.Boolean(string = "Sold", default = False)
+    cancel = fields.Boolean(string = "Cancel", default = False)
+
+    #depends method to calculate total area based on garden area and living area
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+    
+    #depends method to calculate best offer among all other offers
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for record in self:
+            price_list = record.offer_ids.mapped('price')
+            if len(price_list) > 0:
+                record.best_price = max(record.offer_ids.mapped('price'))
+            else:
+                record.best_price = 0
+
+    #onchange method to give default garden area and garden orientation if garden is selected
+    @api.onchange("garden")
+    def _onchange_garden_details(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = ''
+            self.garden_orientation = ''
+    
+    #method to sold property 
+    def action_property_sold(self):
+        if self.status == 'cancelled':
+            raise exceptions.UserError("Cancelled properties cannot be sold.")
+        elif self.status == 'sold':
+            raise exceptions.UserError("Already sold.")
+        else:
+            self.sold = True
+            self.status = 'sold'
+        return
+        
+    #method to cancel property
+    def action_property_cancel(self):
+        if self.status == 'sold':
+            raise exceptions.UserError("Sold properties cannot be cancelled.")
+        if self.status == 'cancelled':
+            raise exceptions.UserError("Already cancelled.")
+        else:
+            self.cancel = True
+            self.status = 'cancelled'
+        return
+
+    #sql constraints to check that expected and selling price is not a negative number
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price >= 0)', 'A property expected price must be strictly positive'),
+        ('check_selling_price', 'CHECK(selling_price >= 0)', 'A property expected price must be positive')
+    ]
+
```

Comment:
```suggestion
    #sql constraints to check that expected and selling price is not a negative number
    _sql_constraints = [
        ('check_expected_price', 'CHECK(expected_price >=0)', 'A property expected price must be strictly positive'),
        ('check_selling_price', 'CHECK(selling_price >= 0)', 'A property expected price must be positive')
    ]
```
Move this below the field declaration